framework/src/media/audio/audio_manager.c

```diff
@@ -159,6 +161,31 @@ static audio_manager_result_t set_audio_volume(audio_card_type_t type, uint8_t v
 /****************************************************************************
  * Private Functions
  ****************************************************************************/
+static audio_manager_result_t init_audio_stream(audio_card_type_t card_type)
+{
+	int i;
+	int max_card_num = CONFIG_AUDIO_MAX_INPUT_CARD_NUM;
+	audio_card_info_t *card_info = g_audio_in_cards;
+
+	if (card_type == OUTPUT) {
+		max_card_num = CONFIG_AUDIO_MAX_OUTPUT_CARD_NUM;
+		card_info = g_audio_out_cards;
+	}
+
+	for (i=0; i < max_card_num; i++) {
+		pthread_mutex_init(&(card_info[i].card_mutex), NULL);
+		medvdbg("The mutex of %s card %d is initialized\n", (card_type == OUTPUT ? "output" : "input"), i);
+	}
+
+	/* check actual cards and try to find all cards if there's no one */
+	if (find_audio_card_if_empty(card_type) != AUDIO_MANAGER_SUCCESS) {
```

Comment:
find_audio_card_if_empty has added to initialize anytime because we didn't use init function.
Now we decided init when it created, we can merge find_audio_card_if_empty && find_audio_card to find_audio_card.

Reply:
Ok. I'll modify.